packages/react-components/src/components/ImageGallery.tsx

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  DefaultButton,
+  IModalStyleProps,
+  IModalStyles,
+  IOverlayStyleProps,
+  IOverlayStyles,
+  IStyle,
+  IStyleFunctionOrObject,
+  Icon,
+  IconButton,
+  Layer,
+  Modal,
+  Stack,
+  mergeStyles
+} from '@fluentui/react';
+
+import React from 'react';
+import { BaseCustomStyles } from '../types';
+import {
+  cancelIcon,
+  closeButtonStyles,
+  controlBarContainerStyle,
+  downloadButtonStyle,
+  downloadIcon,
+  downloadIconStyle,
+  focusTrapZoneStyle,
+  headerStyle,
+  imageStyle,
+  overlayStyles,
+  scrollableContentStyle,
+  titleBarContainerStyle,
+  titleStyle
+} from './styles/ImageGallery.style';
+import { useTheme } from '../theming/FluentThemeProvider';
+import { isDarkThemed } from '../theming/themeUtils';
+
+/**
+ * Fluent styles for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryStylesProps extends BaseCustomStyles {
+  /** Styles for the ImageGallery Modal. */
+  modal?: IStyleFunctionOrObject<IModalStyleProps, IModalStyles>;
+  /** Styles for the ImageGallery Modal overlay. */
+  overlay?: IStyleFunctionOrObject<IOverlayStyleProps, IOverlayStyles>;
+  /** Styles for the ImageGallery header bar. */
+  header?: IStyle;
+  /** Styles for the ImageGallery titleBar container. */
+  titleBarContainer?: IStyle;
+  /** styles for the title label */
+  title?: IStyle;
+  /** Styles for the ImageGallery controlBar container. */
+  controlBarContainer?: IStyle;
+  /** Styles for the download button. */
+  downloadButton?: IStyle;
+  /** Styles for the close modal icon. */
+  closeIcon?: IStyle;
+  /** Styles for the image. */
+  image?: IStyle;
+}
+
+/**
+ * Props for {@link ImageGallery}.
+ *
+ * @beta
+ */
+export interface ImageGalleryImageProps {
+  imageUrl: string;
+  downloadFileName: string;
+  altText?: string;
```

Comment:
what would alt text be normally? the image title?

Reply:
> what would alt text be normally? the image title?

Yes I think it's quite common to use the image title as altText. However, in our case, the image title will be the sender's display name and we don't want to use that as the altText

Reply:
i see, so we won't be setting any alt text then?

Reply:
> i see, so we won't be setting any alt text then?

Good point. Teams web app set it to "image" for inline images. I'll follow that behaviour.